project/gen2.scala

```diff
@@ -175,6 +175,26 @@ object gen2 {
         io.transact(xa).unsafePerformIO
       },
 
+      enum("GcalBaselineType") {
+        type GcalBaselineTypeRec = Record.`'tag -> String`.T
```

Comment:
Right, this is the code generation for `ENUM` types in the database.  We have a second kind of enumeration that are based on normal database tables.  For example `F2Disperser` is one of these.  The code generation is slightly different for these but the resulting Scala code is similar.

The distinction is a bit blurry but I am using `ENUM` types for concepts that shouldn't change like Night or Day baseline calibration.  Table-based enumerations are used for instrument features that can be obsoleted or extended with new values in the future.  I'm not sure though whether to keep that distinction going forward or to just always use the table-based enumerations.  As you noticed, the `ENUM` types require explicit casting.  The table-based enums on the other hand lead to text columns with constraints on the value.